dashboard/src/modules/components/OverviewComponent/NewRunsComponent.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { useState } from "react";
+import "./index.less";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  TableComposable,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+  InnerScrollContainer,
+  OuterScrollContainer,
+  ActionsColumn,
+  ExpandableRowContent,
+} from "@patternfly/react-table";
+import {
+  deleteDataset,
+  updateDataset,
+  setRows,
+  setSelectedRuns,
+} from "actions/overviewActions";
+import { RenderPagination } from "./common-component";
+
+const NewRunsComponent = () => {
+  const dispatch = useDispatch();
+  const { newRuns, initNewRuns, selectedRuns } = useSelector(
+    (state) => state.overview
+  );
+  const loginDetails = useSelector((state) => state.userAuth.loginDetails);
+
+  const [perPage, setPerPage] = useState(5);
+  const [page, setPage] = useState(1);
+
+  const onSetPage = (_evt, newPage, _perPage, startIdx, endIdx) => {
+    setPage(newPage);
+    dispatch(setRows(newRuns.slice(startIdx, endIdx)));
+  };
+
+  const perPageOptions = [
+    { title: "5", value: 5 },
+    { title: "10", value: 10 },
+    { title: "20", value: 20 },
+  ];
+  const onPerPageSelect = (_evt, newPerPage, newPage, startIdx, endIdx) => {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    dispatch(setRows(newRuns.slice(startIdx, endIdx)));
+  };
+  const columnNames = {
+    result: "Result",
+    endtime: "Endtime",
+  };
+
+  /* Selecting */
+  const areAllRunsSelected =
+    newRuns?.length > 0 && newRuns?.length === selectedRuns?.length;
+  const selectAllRuns = (isSelecting) => {
+    dispatch(setSelectedRuns(isSelecting ? newRuns.map((r) => r) : []));
+  };
+  const onSelectRuns = (repo, __rowIndex, isSelecting) => {
+    const otherSelectedRuns = selectedRuns.filter(
+      (r) => r.resource_id !== repo.resource_id
+    );
+    const c = isSelecting ? [...otherSelectedRuns, repo] : otherSelectedRuns;
+    dispatch(setSelectedRuns(c));
+  };
+  const isRowSelected = (run) =>
+    selectedRuns.filter((item) => item.name === run.name).length > 0;
```

Comment:
Should this be using `includes()` instead of `filter()`?  (I.e., how is this situation different from the one in `isRepoExpanded()` at line 112?)

Reply:
In `isRepoExpanded()`, the expanded run names are stored in an array so used `includes()`
In `isRowSelected()`,  the selected runs is an array of object so using `filter()`